Tribler/Test/Core/test_sqlitecachedb.py

```diff
@@ -127,6 +127,23 @@ def execute(sql):
         sqlite_test_2.execute = execute
         sqlite_test_2.initialize()
 
+    @blocking_call_on_reactor_thread
+    def test_integrity_check_triggerd(self):
+        def do_integrity_check(expected=False):
+            self.assertTrue(expected, "Integrity check was not expected")
+
```

Comment:
Please set `do_integrity_check.called = False` here and then set `do_integrity_check.called = True` inside `def do_integrity_check()`. Then you can do an assert on whether it was called or not externally.